projects/sandbox/mdc/mdc/main.py

```diff
@@ -0,0 +1,174 @@
+            hosting the indicated model is running
+        model_name:
+            The name of the model to which to make
+            inference requests
+        data_dir:
+            Directory containing input files representing
+            timeseries on which to perform inference.
+            Each HDF5 file in this directory will be used
+            for inference.
+        write_dir:
+            Directory to which to save raw and locally
+            integrated network outputs.
+        sample_rate:
+            Rate at which input timeseries data has
+            been sampled.
+        inference_sampling_rate:
+            The rate at which to sample windows for
+            inference from the input timeseries.
+            Corresponds to the sample rate of the
+            output timeseries.
+        batch_size:
+            The number of subsequent windows to
+            include in a single batch of inference.
+        window_length:
+            Length of the window over which network
+            outputs should be locally integrated,
+            specified in seconds.
+        max_shift:
+            The maximum shift value across all runs
+            in the timeslide analysis that this run
+            is a part of. This helps keep all output
+            timeseries the same length.
+        throughput:
+            Rate at which to make requests, in units
+            of seconds of data per second `[s' / s]`.
+        sequence_id:
+            Identifier to assign to all the sequences
+            of inference requests in this run to match up
+            with a corresponding snapshot state on the
+            inference server.
+        model_version:
+            Version of the model from which to request
+            inference. Default value of `-1` indicates
+            the latest available version of the model.
+        log_file:
+            File to which to write inference logs.
+        verbose:
+            Flag controlling whether logging verbosity
+            is `DEBUG` (`True`) or `INFO` (`False`)
+    """
+    output_dir.mkdir(exist_ok=True, parents=True)
+    configure_logging(log_file, verbose)
+
+    callback = Callback(
+        id=sequence_id,
+        sample_rate=inference_sampling_rate,
+        batch_size=batch_size,
+        integration_window_length=integration_window_length,
+        cluster_window_length=cluster_window_length,
+        fduration=fduration,
+    )
+
+    # TODO: what's the best place to infer this info?
+    # Obviously in the data loader, but how to communicate
+    # to the batcher/callback?
+    with h5py.File(data_dir / "background.hdf", "r") as f:
+        size = len(f[ifos[0]][str(start)])
+    duration = size / sample_rate
+    end = start + duration
+
+    logging.info(f"Connecting to server at {ip}:8001")
+    client = InferenceClient(
+        f"{ip}:8001", model_name, model_version, callback=callback
+    )
+    chunk_size = int(chunk_size * sample_rate)
```

Comment:
So `chunk_size` is given in seconds? In `infer`, it looks like it refers to a number of indices

Reply:
Ah, nvm, I see where that conversion happens now